daemon/media_socket.c

```diff
@@ -911,40 +911,84 @@ int __get_consecutive_ports(GQueue *out, unsigned int num_ports, unsigned int wa
 }
 
 /* puts a list of "struct intf_list" into "out", containing socket_t list */
-int get_consecutive_ports(GQueue *out, unsigned int num_ports, const struct logical_intf *log,
-		const str *label)
+int get_consecutive_ports(GQueue *out, unsigned int num_ports, struct call_media *media)
 {
 	GList *l;
 	struct intf_list *il;
 	const struct local_intf *loc;
+	const struct logical_intf *log = media->logical_intf;
+	const sockfamily_t *desired_family = media->desired_family;
+	const str *label = &media->call->callid;
 
+	/*
+	// debug locals of logical incerface
+	char ip[100];
 	for (l = log->list.head; l; l = l->next) {
 		loc = l->data;
+		inet_ntop(loc->spec->local_address.addr.family->af, &loc->spec->local_address.addr.u, ip, sizeof(ip));
+		ilog(LOG_DEBUG, "XXXXXXXXXX IP: %s", ip);
+	}
+	ilog(LOG_DEBUG, "");
+	*/
 
-		il = g_slice_alloc0(sizeof(*il));
-		il->local_intf = loc;
-		g_queue_push_tail(out, il);
-		if (G_LIKELY(!__get_consecutive_ports(&il->list, num_ports, 0, loc->spec, label))) {
-			// success - found available ports on local interfaces, so far
-			continue;
-		}
+	if (!rtpe_config.save_interface_ports) {
+		for (l = log->list.head; l; l = l->next) {
+			loc = l->data;
 
-		// error - found at least one local interface with no ports available
-		goto error_ports;
-	}
+			il = g_slice_alloc0(sizeof(*il));
+			il->local_intf = loc;
+			g_queue_push_tail(out, il);
+			if (G_LIKELY(!__get_consecutive_ports(&il->list, num_ports, 0, loc->spec, label))) {
+				// success - found available ports on local interfaces, so far
+				continue;
+			} else {
+				// fail - did not found available ports on at least one local interface
+				goto error_ports;
+			}
+		}
 
-	return 0;
+		return 0;
 
 error_ports:
-	ilog(LOG_ERR, "Failed to get %d consecutive ports on all locals of logical '"STR_FORMAT"'",
-		num_ports, STR_FMT(&log->name));
+		ilog(LOG_ERR, "Failed to get %d consecutive ports on all locals of logical '"STR_FORMAT"'",
+			num_ports, STR_FMT(&log->name));
 
-	// free all ports alloc'ed so far for the previous local interfaces
-	while ((il = g_queue_pop_head(out))) {
-		free_socket_intf_list(il);
-	}
+		// free all ports alloc'ed so far for the previous local interfaces
+		while ((il = g_queue_pop_head(out))) {
+			free_socket_intf_list(il);
+		}
 
-	return -1;
+		return -1;
+
+	} else {
+		for (l = log->list.head; l; l = l->next) {
+			loc = l->data;
+
+			// check desired family of local interface
+			if (desired_family != loc->spec->local_address.addr.family)  {
```

Comment:
Did you have a chance to see if this test is actually necessary? Because I think it should not ever happen. But we can leave it in here for now.

Reply:
Yes, I've been looking into it and I consider it is needed because:
1. local interfaces from a logical interface are indeed ordered cosidering the desired family, however, it won't contain only local interfaces of that desired family but both families e.g. (IP6 IP6 IP4 IP4, not just IP6 IP6)
2. If IP6 is desired, and no local interface has available ports, I'd like __not__ to bind IP4 ones, but just give error

Reply:
Fair enough, thanks for explanation